app/src/main/java/com/example/traders/presentation/watchlist/WatchListRepository.kt

```diff
@@ -1,166 +1,167 @@
-//            low = low,
-//            open = open,
-//            priceChange = priceChange,
-//            priceChangePercent = priceChangePercent
-//        )
-//    }
-//
-//    private fun List<BinanceDataItem>.applyFavourites(favouriteList: List<FavouriteCrypto>?): List<BinanceDataItem> {
-//        return map { item ->
-//            item.copy(
-//                isFavourite = favouriteList?.any { it.symbol == item.symbol } == true
-//            )
-//        }
-//    }
-//
-//    //    I will have to start and pause collecting state
-//    // It collects message emitted from websocket sharedFlow and updates list item by reassigning BinanceDataItem to new value
-//    suspend fun startCollectingBinanceTickerData() {
-//        webSocketClient.state.collect { tickerData ->
-//            val indexOfCryptoDataToUpdate = _binanceCryptoList.value.indexOfFirst {
-//                it.symbol == tickerData.symbol
-//            }
-//
-//            if (indexOfCryptoDataToUpdate != -1) {
-//                _binanceCryptoList.value = _binanceCryptoList.value.let {
-//                    val updatedList = it.toMutableList()
-//                    val itemToUpdate =
-//                        tickerData.toBinanceDataItem(updatedList[indexOfCryptoDataToUpdate].isFavourite)
-//                    if (itemToUpdate != null) {
-//                        updatedList[indexOfCryptoDataToUpdate] = itemToUpdate
-//                    }
-//                    updatedList
-//                }
-//            }
-//        }
-//    }
-//
-//    // Converts PriceTickerData tp Binance24DataItem
-//    private fun PriceTickerData?.toBinanceDataItem(isFavourite: Boolean): BinanceDataItem? {
-//        if (this == null) return null
-//        return BinanceDataItem(
-//            symbol = symbol.replace("USDT", ""),
-//            last = last,
-//            high = high,
-//            low = low,
-//            open = open,
-//            priceChange = priceChange,
-//            priceChangePercent = priceChangePercent,
-//            isFavourite = isFavourite
-//        )
-//    }
-//}
\ No newline at end of file
+
+import com.example.traders.database.CryptoDatabaseDao
+import com.example.traders.database.FavouriteCrypto
+import com.example.traders.database.PreferancesManager
+import com.example.traders.database.SortOrder
+import com.example.traders.network.BinanceApi
+import com.example.traders.network.repository.enumContains
+import com.example.traders.utils.MappingUtils.enumConstantNames
+import com.example.traders.database.FixedCryptoList
+import com.example.traders.network.models.binance24HourData.Binance24DataItem
+import com.example.traders.network.models.binance24HourData.BinanceDataItem
+import com.example.traders.network.models.binance24hTickerData.PriceTickerData
+import com.example.traders.network.webSocket.BinanceWSClient
+import kotlinx.coroutines.flow.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```

Comment:
Singleton is expensive class as it lives as long as application lives. It should be avoided if possible.